vault/ui_custom_message.go

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	UICustomMessageKey string = "/custom-messages"
+
+	MaximumCustomMessageCount int = 100
```

Comment:
Question: is there a reason we settled on this number? Why not more?

Reply:
Arbitrarily picked. I've done some benchmarking on the `(*UIConfig).ListCustomMessages` method with this number and it currently takes on the order of 200-250 microseconds to complete when 100 custom messages are present.

I'd like to revisit this once this PR is merged (and the UI team is unblocked) to do a bit more benchmarking to find a good number that doesn't hamper performance too much.